src/mcts/search.cc

```diff
@@ -1510,6 +1510,21 @@ void SearchWorker::DoBackupUpdateSingleNode(
     n->FinalizeScoreUpdate(v / (1.0f + params_.GetShortSightedness() * depth),
                            d, m, node_to_process.multivisit);
 
+    // Flatten policy if we've visited the node enough times. We efficiently
+    // sample every 2^13 visits as an approximation to 10k visits. This is not
+    // exact with multivisits maybe skipping samples, but this is good enough.
+    const auto visits = n->GetN();
+    const auto ratio = params_.GetPolicyFlattenRatio();
+    if (ratio && !(visits & (1 << 13) - 1)) {
```

Comment:
For those curious…
```asm
        movl    64(%rbx), %ecx
        testl   $8191, %ecx             ## imm = 0x1FFF
        jne     LBB56_22
## %bb.12:                              ##   in Loop: Header=BB56_11 Depth=1
        movq    96(%r13), %rax
        vmovss  64(%rax), %xmm0         ## xmm0 = mem[0],zero,zero,zero
        vucomiss        LCPI56_2(%rip), %xmm0
        jne     LBB56_13
        jnp     LBB56_22
```
Looks like the optimizer says the visits check against 8191 is better to do first than the non-zeroness of ratio.